### PR TITLE
hello segen

### PR DESCRIPTION
**Pull request by PolynomialDivision**

### CODE REVIEW

**Review by Noki — CHANGES_REQUESTED** [review]
small changes needed

File: group_vars/location_segen/snmp.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    snmp_profile: airos_8
+  - hostname: segen-mabb
+    address: 10.31.6.12
+    snmp_profile: airos_8
+  - hostname: segen-sw
+    address: 10.31.6.13
+    snmp_profile: airos_8
+  - hostname: segen-sw
+    address: 10.31.6.14
+    snmp_profile: airos_8
```
Comment (Noki) [inline group_vars/location_segen/snmp.yml:18]: Entry is a duplicate. Following IPs need to be reduced by one

**Review by Noki — CHANGES_REQUESTED** [review]
small changes needed

File: group_vars/location_segen/networks.yml [hunk 2]
```diff
@@ -0,0 +1,181 @@
+---
+# IPv4 10.31.6.0/24
+# IPv6 2001:bf7:760:1000::/56
+
+# DHCP: 10.31.6.128/25
```
Comment (Noki) [inline group_vars/location_segen/networks.yml:5]: should be 10.31.66.0/24 now

File: roles/cfg_openwrt/templates/common/config/network.j2 [hunk 3]
```diff
@@ -39,6 +39,8 @@ config interface '{{ name }}'
     {% if network.get('enforce_client_isolation') and role == 'corerouter' and
           not bridge_needed %}
 	option macaddr '02:00:00:00:00:01'
+    {% elif network.get('mac_override') and role == 'corerouter' %}
```
Comment (pmelange) [inline roles/cfg_openwrt/templates/common/config/network.j2:42]: Maybe you should check to see if role == uplink_gateway too

**Review by pmelange — COMMENTED** [review]
just a small change req

File: group_vars/location_segen/airos_dfs_reset.yml [hunk 4]
```diff
@@ -0,0 +1,23 @@
+---
+
+airos_dfs_reset:
+  - name: "segen-f2a"
+    target: "10.31.6.11"
+    username: "ubnt"
+    password: "file:/root/pwd"
+    daytime_limit: "2-7"
+  - name: "segen-mabb"
```
Comment (spolack) [inline group_vars/location_segen/airos_dfs_reset.yml:9]: please put a blank line before all items

File: group_vars/location_segen/networks.yml [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+
+      # nf
+      segen-n-nf-2ghz: 24
+      segen-o-nf-2ghz: 25
+      segen-s-nf-2ghz: 26
+      segen-w-nf-2ghz: 27
+
+  - vid: 10
+    role: dhcp
+    prefix: 10.31.66.0/24
```
Comment (spolack) [inline group_vars/location_segen/networks.yml:47]: /26  or even /27 should be sufficient, given the maximum simultanous clients were 10?

File: group_vars/location_segen/networks.yml [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+    ipv6_subprefix: -8
+
+
+  # 11s mesh 2ghz
+  - vid: 201
+    role: mesh
+    name: mesh_11s_n2
+    prefix: 10.31.6.72/32
+    ipv6_subprefix: -9
+    mesh_metric: 1024
```
Comment (spolack) [inline group_vars/location_segen/networks.yml:115]: maybe put also a default LQM of 0.3 on all 11s interfaces to prefer real links over "wild" mashing

File: group_vars/location_segen/networks.yml [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+    mesh_ap: segen-n-nf-2ghz
+    mesh_radio: 11g_standard
+    mesh_iface: mesh
+    mac_override: f0:9f:c2:0f:79:31
```
Comment (spolack) [inline group_vars/location_segen/networks.yml:119]: why are you doing this?  dont know this feature. Mac override works differently:
See following links for reference
https://github.com/freifunk-berlin/bbb-configs/blob/29a001b233854b03cdf0a16724b971a706aab8b3/roles/cfg_openwrt/templates/common/config/network.j2#L80
https://github.com/freifunk-berlin/bbb-configs/blob/29a001b233854b03cdf0a16724b971a706aab8b3/host_vars/magda-ap1/base.yml#L7

File: group_vars/location_segen/networks.yml [hunk 8]
```diff
@@ -0,0 +1,188 @@
+---
+# IPv4 10.31.6.0/24
```
Comment (spolack) [inline group_vars/location_segen/networks.yml:2]: Why two /24? Please stop wasting IPs

File: group_vars/location_segen/networks.yml [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+# IPv4 10.31.66.0/24
+# IPv6 2001:bf7:760:1000::/56
+
+# DHCP: 10.31.66.0/24
+# MGMT: 10.31.6.0/26
+# MESH: 10.31.6.64/26
```
Comment (spolack) [inline group_vars/location_segen/networks.yml:8]: /26 means 64 Mesh interfaces??

File: group_vars/location_segen/networks.yml [hunk 10]
```diff
@@ -0,0 +1,188 @@
+---
+# IPv4 10.31.6.0/24
+# IPv4 10.31.66.0/24
+# IPv6 2001:bf7:760:1000::/56
+
+# DHCP: 10.31.66.0/24
+# MGMT: 10.31.6.0/26
```
Comment (spolack) [inline group_vars/location_segen/networks.yml:7]: /26 means 61 Devices??

File: group_vars/location_segen/networks.yml [hunk 11]
```diff
@@ -0,0 +1,188 @@
+---
+# IPv4 10.31.6.0/24
+# IPv4 10.31.66.0/24
+# IPv6 2001:bf7:760:1000::/56
+
+# DHCP: 10.31.66.0/24
```
Comment (spolack) [inline group_vars/location_segen/networks.yml:6]: see below

File: group_vars/location_segen/snmp.yml [hunk 12]
```diff
@@ -0,0 +1,39 @@
+---
+
+snmp_devices:
+  - hostname: segen-saarbr
+    address: 10.31.6.10
+    snmp_profile: airos_8
```
Comment (spolack) [inline group_vars/location_segen/snmp.yml:6]: airos 8?

File: roles/cfg_openwrt/templates/common/config/network.j2 (hunk 3 above)
Comment (spolack) [inline roles/cfg_openwrt/templates/common/config/network.j2:42]: why are you implementing this? Please open seperate PR if you wanna introduce a feature

File: roles/cfg_openwrt/templates/common/config/system.j2 [hunk 13]
```diff
@@ -17,7 +17,7 @@ config system
 
 config timeserver 'ntp'
         option enabled '1'
-        option enable_server '0'
+        option enable_server '{{ (role == 'corerouter' or role  == 'uplink_gateway') | int }}'
```
Comment (spolack) [inline roles/cfg_openwrt/templates/common/config/system.j2:20]: this is not related to Segen. Please open a seperate PR

File: roles/cfg_openwrt/templates/corerouter/config/dhcp.j2 [hunk 14]
```diff
@@ -27,6 +27,7 @@ config dhcp 'dhcp_{{ name }}'
 	option leasetime '5m'
 	option start '2'
 	option limit '{{ network['prefix'] | ipaddr('size') -3 }}'
+	list dhcp_option '119,olsr'
```
Comment (spolack) [inline roles/cfg_openwrt/templates/corerouter/config/dhcp.j2:30]: This is not related to segen bringup? Please open a seperate PR, also explaining what would be the advantage with that change

**Review by spolack — CHANGES_REQUESTED** [review]

File: group_vars/location_segen/networks.yml (hunk 5 above)
Comment (pmelange) [reply group_vars/location_segen/networks.yml:47]: Sometimes there are hoffests at the church.  Also, Segen has had this address range since day 1.

```
-         10.31.6.0/24                          R  -   None               None      freifunk@faust2k.net  Segenskirche                                                                      
-       10.36.0.0/16                            R  -   None               None      None                  HNA                                                                               
-         10.36.66.0/24                         R  -   None               None      freifunk@faust2k.net  Segenskirche                                                                      
```

I would prefer to leave it how it is.

File: group_vars/location_segen/networks.yml (hunk 7 above)
Comment (pmelange) [reply group_vars/location_segen/networks.yml:119]: The seperate MAC addresses prevent warning messages about packets coming in on an interface with the same MAC address.  This would happen when two 802.11s devices on the same core system see each other.

The end result is
```
config interface 'mesh_11s_n2'
	option device 'switch0.201'
	option macaddr 'f0:9f:c2:0f:79:31'
	option proto 'static'
	option ipaddr '10.31.6.72/32'
	option ip6addr '2001:bf7:760:10f7::1/128'

config interface 'mesh_11s_o2'
	option device 'switch0.202'
	option macaddr 'f0:9f:c2:0f:79:32'
	option proto 'static'
	option ipaddr '10.31.6.73/32'
	option ip6addr '2001:bf7:760:10f6::1/128'

config interface 'mesh_11s_s2'
	option device 'switch0.203'
	option macaddr 'f0:9f:c2:0f:79:33'
	option proto 'static'
	option ipaddr '10.31.6.74/32'
	option ip6addr '2001:bf7:760:10f5::1/128'

config interface 'mesh_11s_w2'
	option device 'switch0.204'
	option macaddr 'f0:9f:c2:0f:79:34'
	option proto 'static'
	option ipaddr '10.31.6.75/32'
	option ip6addr '2001:bf7:760:10f4::1/128'
```
Which I think is the proper way for it to look.

File: group_vars/location_segen/networks.yml (hunk 8 above)
Comment (pmelange) [reply group_vars/location_segen/networks.yml:2]: These IP's, as stated above, have belonged to Segen since day 1.

File: group_vars/location_segen/networks.yml (hunk 6 above)
Comment (spolack) [reply group_vars/location_segen/networks.yml:115]: maybe 0.4 could be a compromise? :)

File: group_vars/location_segen/networks.yml (hunk 7 above)
Comment (spolack) [reply group_vars/location_segen/networks.yml:119]: This issue is already solved https://github.com/freifunk-berlin/bbb-configs/issues/119

File: group_vars/location_segen/networks.yml (hunk 5 above)
Comment (pmelange) [reply group_vars/location_segen/networks.yml:47]: I respectfully disagree

File: group_vars/location_segen/networks.yml (hunk 6 above)
Comment (pmelange) [reply group_vars/location_segen/networks.yml:115]: 0.4 is a reasonable compromise

File: group_vars/location_segen/networks.yml (hunk 7 above)
Comment (pmelange) [reply group_vars/location_segen/networks.yml:119]: I have no idea if this works and I don't not have the capacity at the moment to test it.

File: group_vars/location_segen/networks.yml (hunk 8 above)
Comment (Noki) [reply group_vars/location_segen/networks.yml:2]: As discussed with all participants: Both /24 IP-Blocks should be kept for now as they were already assigned. This might not be optimal in terms of IPv4 address usage but this should not be a blocker for finalizing and merging this. We can reevaluate dropping the additional block after Corona, once we have more statistics based on the next "Hoffest".

File: group_vars/location_segen/networks.yml (hunk 5 above)
Comment (Noki) [reply group_vars/location_segen/networks.yml:47]: see comment above regarding already assigned IPs.

File: group_vars/location_segen/networks.yml (hunk 7 above)
Comment (Noki) [reply group_vars/location_segen/networks.yml:119]: As discussed with all participants: Test if the fix #119 actually works and remove the overrides if it does. If the warning messages start showing up, keep the overrides and create an issue to fix the underlying problem in general.

File: group_vars/location_segen/snmp.yml (hunk 12 above)
Comment (Noki) [reply group_vars/location_segen/snmp.yml:6]: Please create an SNMP Profile for GIGABEAM LR  and use it.

**Comment by Akira25** [comment]
As requested by @Noki, I'm going to close this, as most feedback was included into #235. That PR presents a working state with the lowest common denominator I guess. As it is already deployed on Segenskirche, it shouldn't be delayed too much, as bbb-configs is meant to be a documentation too.

Please consider submitting and discussing the other stuff in a new PR.
